From 1608342a38731f8a669b203db448576701df4451 Mon Sep 17 00:00:00 2001
From: "Panwar, Ashish" <ashish.panwar@intel.com>
Date: Fri, 18 Mar 2016 16:01:07 +0530
Subject: [PATCH 3956/4301] ASoC: Intel: CNL: Initialize trace buffer window
 for CNL

commit 49d566637830353a3e747e00174b9ca9f6be1ba1 from
git://git@github.com:01org/linux-intel-4.9.git

Initialize address, size of tracing window and write pointers
of each core for CNL platform.

Change-Id: I9febfe5bd1eef76f50f5de170c6c99fc98cdc6d9
Signed-off-by: Panwar, Ashish <ashish.panwar@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/8586
Reviewed-by: Shah, Hardik T <hardik.t.shah@intel.com>
Tested-by: Shah, Hardik T <hardik.t.shah@intel.com>
---
 sound/soc/intel/skylake/cnl-sst.c |   19 +++++++++++++++++++
 1 files changed, 19 insertions(+), 0 deletions(-)

diff --git a/sound/soc/intel/skylake/cnl-sst.c b/sound/soc/intel/skylake/cnl-sst.c
index f1758f4..6ba36f3 100644
--- a/sound/soc/intel/skylake/cnl-sst.c
+++ b/sound/soc/intel/skylake/cnl-sst.c
@@ -39,6 +39,7 @@
 #include "cnl-sst-dsp.h"
 #include "skl-sst-dsp.h"
 #include "skl-sst-ipc.h"
+#include "skl-fwlog.h"
 
 #define FW_ROM_INIT_DONE                0x1
 #define CNL_IPC_PURGE_FW		0x01004000
@@ -50,6 +51,14 @@
 /* Intel HD Audio SRAM Window 0*/
 #define CNL_ADSP_SRAM0_BASE	0x80000
 
+/* Trace Buffer Window */
+#define CNL_ADSP_SRAM2_BASE     0x0C0000
+#define CNL_ADSP_W2_SIZE        0x2000
+#define CNL_ADSP_WP_DSP0        (CNL_ADSP_SRAM0_BASE+0x30)
+#define CNL_ADSP_WP_DSP1        (CNL_ADSP_SRAM0_BASE+0x34)
+#define CNL_ADSP_WP_DSP2        (CNL_ADSP_SRAM0_BASE+0x38)
+#define CNL_ADSP_WP_DSP3        (CNL_ADSP_SRAM0_BASE+0x3C)
+
 /* Firmware status window */
 #define CNL_ADSP_FW_STATUS	CNL_ADSP_SRAM0_BASE
 #define CNL_ADSP_ERROR_CODE	(CNL_ADSP_FW_STATUS + 0x4)
@@ -666,6 +675,8 @@ int cnl_sst_dsp_init(struct device *dev, void __iomem *mmio_base, int irq,
 {
 	struct skl_sst *cnl;
 	struct sst_dsp *sst;
+	u32 dsp_wp[] = {CNL_ADSP_WP_DSP0, CNL_ADSP_WP_DSP1, CNL_ADSP_WP_DSP2,
+				CNL_ADSP_WP_DSP3};
 	int ret;
 
 	cnl = devm_kzalloc(dev, sizeof(*cnl), GFP_KERNEL);
@@ -693,6 +704,14 @@ int cnl_sst_dsp_init(struct device *dev, void __iomem *mmio_base, int irq,
 			CNL_ADSP_W0_UP_SZ, CNL_ADSP_SRAM1_BASE, CNL_ADSP_W1_SZ);
 
 	INIT_LIST_HEAD(&sst->module_list);
+
+	ret = skl_dsp_init_trace_window(sst, dsp_wp, CNL_ADSP_SRAM2_BASE,
+					 CNL_ADSP_W2_SIZE, CNL_DSP_CORES);
+	if (ret) {
+		dev_err(dev, "FW tracing init failed : %x", ret);
+		return ret;
+	}
+
 	ret = cnl_ipc_init(dev, cnl);
 	if (ret)
 		return ret;
-- 
1.7.5.4

